src/status_emitter/json.rs

```diff
@@ -0,0 +1,157 @@
+use super::RevisionStyle;
+use super::StatusEmitter;
+use super::Summary;
+use super::TestStatus;
+use crate::TestOk;
+use crate::test_result::TestResult;
+
+use std::boxed::Box;
+use std::fmt::Debug;
+use std::path::{Path, PathBuf};
+
+use bstr::ByteSlice;
+
+// MAINTENANCE REGION START
+
+// When integrating with a new libtest version, update all emit_xxx functions.
+
+fn emit_suite_end(failed: usize, filtered_out: usize, ignored: usize, passed: usize, status: &String,) {
+    // Adapted from test::formatters::json::write_run_finish().
+    println!(r#"{{ "type": "suite", "event": "{status}", "passed": {passed}, "failed": {failed}, "ignored": {ignored}, "measured": 0, "filtered_out": {filtered_out} }}"#);
+}
+
+fn emit_test_end(name: &String, status: &String, error_output: &String) {
```

Comment:
```suggestion
fn emit_test_end(name: &str, status: &str, error_output: &str) {
```

(clippy CI will likely tell you that anyway ^^)